backend/layers/processing/submissions/app.py

```diff
@@ -34,7 +35,8 @@ def get_business_logic():
     global _business_logic
     if not _business_logic:
         database_provider = DatabaseProvider()
-        _business_logic = BusinessLogic(database_provider, None, None, None, None)
+        uri_provider = UriProvider()
+        _business_logic = BusinessLogic(database_provider, None, None, None, uri_provider)
```

Comment:
I think we need the step function provider as well, its invoked in ingest_dataset